src/india_api/internal/service/resample.py

```diff
@@ -16,6 +16,7 @@ def resample_generation(values: list[ActualPower], internal_minutes: int) -> lis
 
     # resample
     df = df.set_index("Time").resample(f"{internal_minutes}T").mean().dropna()
+    df.loc[df['PowerKW'] < 0, 'PowerKW'] = 0  # Set negative values of PowerKW up to 0
```

Comment:
Do you mind doing something like this 
`df['PowerKW'].clip(lower=0, inplace=True)`,
I think its slightly easier to read